cpp/src/arrow/util/compression-test.cc

```diff
@@ -69,37 +69,39 @@ void CheckCodecRoundtrip(Compression::type ctype, const std::vector<uint8_t>& da
   int64_t actual_size;
   ASSERT_OK(c1->Compress(data.size(), data.data(), max_compressed_len, compressed.data(),
                          &actual_size));
-  compressed.resize(actual_size);
```

Comment:
I'm not sure I understand. Why do you remove this?

Reply:
I use returned sizes below, rather the explicit vector size.   

Snappy appears to use more than the final compressed size of the buffer when compressing which triggers:

https://github.com/google/sanitizers/wiki/AddressSanitizerContainerOverflow

Reply:
You mean when decompressing?

Reply:
No compressing.  The problem is actually below on the second compress round.  This seemed like a cleaner fix then resizing up again before the second compression (but I suppose it prevents sanitizer verification of decompression not consuming more bytes then it should)

Reply:
I think resizing again before the second compression would make the intent clearer.

Reply:
Sounds good, I'll do that

Reply:
@pitrou made the changes.